mne/beamformer/_lcmv.py

```diff
@@ -73,37 +74,110 @@ def _check_one_ch_type(info, picks, noise_cov):
         [_contains_ch_type(info_pick, tt) for tt in ('mag', 'grad', 'eeg')]
     if sum(ch_types) > 1 and noise_cov is None:
         raise ValueError('Source reconstruction with several sensor types '
-                         'requires a set of noise covariance matrices to be '
+                         'requires a noise covariance matrix to be '
                          'able to apply whitening.')
 
 
-@verbose
-def _apply_lcmv(data, info, tmin, forward, reg, noise_cov=None,
-                data_cov=None, label=None, picks=None, pick_ori=None,
-                rank=None, weight_norm=None, max_ori_out=None, verbose=None):
-    """LCMV beamformer for evoked data, single epochs, and raw data."""
-    is_free_ori, ch_names, proj, vertno, G = \
-        _prepare_beamformer_input(info, forward, label, picks, pick_ori)
+def _pick_channels_spatial_filter(ch_names, filters):
+    """Return data channel indices to be used with spatial filter.
 
-    if max_ori_out == 'abs':
-        warn('max_ori_out and the return of absolute values is deprecated and '
-             'will be removed in 0.16. Set it to "signed" to remove this '
-             'warning, this will return signed time series.',
-             DeprecationWarning)
+    Unlike ``pick_channels``, this respects the order of ch_names.
+    """
+    ch_names = set(ch_names)
```

Comment:
Yes but it has no meaning to use a set here. @agramfort did not strictly intend this I suppose. For looking up membership uniqueness is not mandatory and channel names are anyways unique by construction in MNE. You can just remove this line without removing functionality or benefits.
Also call the running index ii or ii_ch or something like this.